celltype_annotation/analysis/SingleR-combining-refs.Rmd

```diff
@@ -0,0 +1,203 @@
+  dplyr::mutate(reference = forcats::fct_recode(as.factor(reference),
+    "BlueprintEncodeData" = "1", 
+    "HumanPrimaryCellAtlasData" = "2",
+    "DatabaseImmuneCellExpressionData" = "3",
+    "MonacoImmuneData" = "4"
+   ))
+```
+
+The first thing I looked at was the labels that end up getting assigned and which reference was used to assign those labels. 
+
+```{r, fig.height = 5}
+# only pull out the top 10 labels 
+top_label_counts_df <- result_df |>
+  dplyr::mutate(top_labels = forcats::fct_lump_n(pruned.labels, 10)) |>
+  dplyr::count(top_labels, reference)
+
+# same barchart but with just the top 10 labels
+ggplot(top_label_counts_df, aes(x = reference, fill = top_labels, y = n)) +
+  geom_bar(position = "stack", stat = "identity") +
+  labs(x = "",
+       y = "Number of cells",
+       label = "Cell type") +
+  guides(x = guide_axis(angle = 90))
+```
+
+It looks like most of the labels are coming from monocyte progenitors in BlueprintEncodeData reference with a few cells being assigned to other labels present in the other datasets.
+
+Below, I looked at how the labels looked when using all references vs. single references. 
+When running SingleR with multiple references, first DE genes are identified within each reference and labels assigned within each reference and then statistics are merged across references (I'm still slightly unclear on exactly what that means).
+
+```{r, fig.height=5}
+# get all the labels from combined and each individual reference
+all_results <- result_df |> 
+  dplyr::select(ends_with("pruned.labels")) |>
+  tidyr::pivot_longer(cols = everything(), 
+                      values_to = "label.ont",
+                      names_to = "reference") |> 
+  # rename final assigned labels as combined and remove some extra strings
+  dplyr::mutate(reference = ifelse(reference == "pruned.labels", "combined", reference),
+                reference = stringr::str_remove_all(reference, "orig.results.|.pruned.labels")) |>
+  # grab top 10 and relevel for ordering |>
+  dplyr::mutate(top_labels = forcats::fct_lump_n(label.ont, 10), 
+                reference = factor(reference, 
+                                   levels = c("combined", 
+                                              "BlueprintEncodeData", 
+                                              "HumanPrimaryCellAtlasData",
+                                              "DatabaseImmuneCellExpressionData",
+                                              "MonacoImmuneData"))) |>
+  dplyr::count(top_labels, reference)
+
+# create a barchart to look at assignments between combined vs. single refs 
+ggplot(all_results, aes(x = reference, fill = top_labels, y = n)) +
+  geom_bar(position = "stack", stat = "identity")  +
+  labs(x = "",
+       y = "Number of cells",
+       label = "Cell type") +
+  guides(x = guide_axis(angle = 90))
+  
+  
+```
+
+It looks like the combined references look the most similar to the BlueprintEncodeData. 
```

Comment:
This plot seems to me to be possibly the most useful one here. I am particularly interested by just comparing the blueprint to HPCA data, which seem to actually disagree on the labels "common myeloid progenitor" and "granulocyte monocyte progenitor" for a significant portion of cells. Blueprint "wins" here, but I honestly don't know what to make of that: is it a better reference, or just better matched to this dataset?

I think it might be interesting to plot a matrix of comparison between the the combined data and each reference, ideally some kind of annotation of the relationships among labels: when two references disagree, is there a parent/child relationship between the labels? We'd have to dive further into the ontology to create such a plot, so I think it is out of scope for this notebook. We would want to spend some time sketching out designs before trying to actually generate the plot, so this is a long-term thought.